packages/jaeger-ui/src/components/DeepDependencyGraph/HopsSelector/Selector.tsx

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { PureComponent } from 'react';
+import { Popover } from 'antd';
+import cx from 'classnames';
+
+import { EStream, EFullness, THop } from './types';
+
+import './Selector.css';
+
+type TProps = {
+  furthestDistance: number;
+  furthestFullness: EFullness;
+  handleClick: (distance: number, direction: EStream) => void;
+  hops: THop[];
+  stream: EStream;
+};
```

Comment:
I don't see why hops need `checkedScope` or `totalScope` instead of `distance`
Also, `checked` as a type name is pretty tied to one of its use cases, not what it actually measures.
I updated the Enum to `ECheckedStatus`, but `THop` still has the keys `distance` and `fullness`.

Reply:
Also, the `currentScope` / `totalScope` on the `Selector` was part of the original ticket, but I think its additional complexity warrants splitting it into its own ticket.
I don't think there would be overlap in implementation, and it will much more closely reflect the work for show/hide children.